metricbeat/module/kafka/consumergroup/query.go

```diff
@@ -38,51 +39,55 @@ func fetchGroupInfo(
 
 	debugf("known consumer groups: ", groups)
 
+	assignments, err := fetchGroupAssignments(b, groups)
+	if err != nil {
+		logp.Err("failed to fetch kafka group assignments: %v", err)
+		return err
+	}
+	if len(assignments) == 0 {
+		return nil
+	}
+
 	wg := sync.WaitGroup{}
-	results := make(chan result, len(groups))
-	for _, group := range groups {
-		group := group
+	results := make(chan result, len(assignments))
+
+	waiting := 0
+	for group, topics := range assignments {
+		_partitions := make(map[string][]int32)
```

Comment:
no underscore please.

in beats codebase we prefer to use make/new only if there is not other syntax pattern. Use

```
partitions = map[string][]int32{}
```

How about (reduce key lookups and appends/alloc to slice):

```
  queryTopics := map[string][]int32{}
  for topic, partitions := range topics {
    if topicFilter != nil && !topicFilter(topic) {
      continue
    }

    // copy partition ids
    ids, i := make([]int32, len(partitions)), 0
    for partition := range partitions {
      ids[i], i = partition, i+1
    }
    queryTopics[topic] = ids
  }

  if len(queryTopics) == 0 {
    continue
  }
 
  ...
```

Reply:
fixed